packages/script-editor/src/ScriptEditorWidgetFactory.tsx

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2018-2022 Elyra Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { CodeEditor, IEditorServices } from '@jupyterlab/codeeditor';
+import {
+  ABCWidgetFactory,
+  DocumentRegistry,
+  DocumentWidget
+} from '@jupyterlab/docregistry';
+import { FileEditor } from '@jupyterlab/fileeditor';
+
+import { ScriptEditor } from './ScriptEditor';
+
+/**
+ * A widget factory for script editors.
+ */
+export class ScriptEditorWidgetFactory extends ABCWidgetFactory<
```

Comment:
two comments on this (both non-blocking):

1. Why the name change from `ScriptEditorFactory` to `ScriptEditorWidgetFactory`?

2. Why move this into a separate file?

Reply:
1. The name has changed to make it more clear that it is a widget factory class.
2. It has moved to its own file because the the editor is independent of its factory (imo this should have been done before). Makes it easier to maintain (the `ScriptEditor` class will become more complex with the debugger), follows common conventions and doesn't expose to `ScriptEditor` things it doesn't need to know about from certain libraries.